doc/languages-frameworks/python.section.md

```diff
@@ -1362,8 +1363,53 @@ work with any of the [existing hooks](#setup-hooks).
 
 The `pythonRelaxDepsHook` has no effect on build time dependencies, such as
 those specified in `build-system`. If a package requires incompatible build
-time dependencies, they should be removed in `postPatch` through
-`substituteInPlace` or similar.
+time dependencies, consider using `relaxBuildSystem` or `removeBuildSystem`.
+
+#### Using relaxBuildSystemRequiresHook {#using-relaxbuildsystemrequireshook}
+
+`relaxBuildSystemRequiresHook` has the same function as `pythonRelaxDepsHook`,
+but it affects `build-system` instead of `dependencies`.
+
+For example, given the following `pyproject.toml` file:
+
+```toml
+[build-system]
+requires = [
+  "pkg1<1.0",
+  "pkg2",
+  "pkg3>=1.0,<=2.0",
+]
+```
+
+we can do:
+
+```nix
+{
+  relaxBuildSystem = [
+    "pkg1"
+    "pkg3"
+  ];
+  removeBuildSystem = [
+    "pkg2"
+  ];
+}
+```
+
+which would result in the following `pyproject.toml` file:
+
+```toml
+[build-system]
+requires = [
+  "pkg1",
+  "pkg3",
+]
+```
+
+In general you should always use `relaxBuildSystem`, because `removeBuildSystem`
+will cause silent build errors. However `removeBuildSystem` may
+still be useful in exceptional cases, and also to remove dependencies wrongly
+declared by upstream (for example, declaring `pytest` as a build dependency
+instead of a test dependency).
```

Comment:
I think also mentioning `numpy_2` here is a good idea, since they [explicitly reccomend](https://numpy.org/devdocs/dev/depending_on_numpy.html#numpy-2-0-specific-advice) that you add `numpy>=2` to `requires` even when compatible with v1